### PR TITLE
Sign up with full name and prompt to enter weight

### PR DESCRIPTION
**Pull request by tsrandrei**
The PR covers work for:

- https://trello.com/c/Uw4hpoD2
- https://trello.com/c/CzKCx0gl

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -1,9 +1,13 @@
...
 require 'rails_helper'
 
-RSpec.describe User, type: :model do
-  pending "add some examples to (or delete) #{__FILE__}"
+RSpec.describe User do
+  describe 'validations' do
+    it { is_expected.to validate_presence_of :email }
+    it { is_expected.to validate_presence_of :first_name }
+    it { is_expected.to validate_presence_of :last_name }
+    # case sensitivity is an issue for Devise, even if downcase the email
```
Comment (lshepstone) [inline spec/models/user_spec.rb]: Let's remove this comment as well, it doesn't tell us anything more than what the code does (which means it's good code 😉)

File: app/assets/stylesheets/style.css [hunk 2]
```diff
@@ -44,7 +44,7 @@ h1, h2, h3, h4, h5, h6 {
     width: 100%;
     height: 100%;
     overflow: visible;
-    background: #fff url("../img/preloader.svg") no-repeat center center;
+    /*background: #fff url("../img/preloader.svg") no-repeat center center;*/
```
Comment (lshepstone) [inline app/assets/stylesheets/style.css]: No need for commented out code here

File: app/assets/stylesheets/style.css [hunk 3]
```diff
@@ -140,7 +140,7 @@ h1, h2, h3, h4, h5, h6 {
 #hero {
     width: 100%;
     height: 100vh;
-    background: url(../img/hero-bg.jpg) top center;
+    /*background: url(../img/hero-bg.jpg) top center;*/
```
Comment (lshepstone) [inline app/assets/stylesheets/style.css]: No need for commented out code here, let's remove it

File: app/assets/stylesheets/style.css [hunk 4]
```diff
@@ -671,7 +671,7 @@ body.mobile-nav-active #mobile-nav-toggle {
 /* Call To Action Section
 --------------------------------*/
 #call-to-action {
-    background: linear-gradient(rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6)), url(../img/call-to-action-bg.jpg) fixed center center;
+    /*background: linear-gradient(rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6)), url(../img/call-to-action-bg.jpg) fixed center center;*/
```
Comment (lshepstone) [inline app/assets/stylesheets/style.css]: No need for commented out code here, let's remove it

File: app/assets/stylesheets/style.css [hunk 5]
```diff
@@ -996,7 +996,7 @@ body.mobile-nav-active #mobile-nav-toggle {
 
 #main-layout .main-container .background {
     min-height: 300px;
-    background: url(../img/about-img.jpg) center top no-repeat;
+    /*background: url(../img/about-img.jpg) center top no-repeat;*/
```
Comment (lshepstone) [inline app/assets/stylesheets/style.css]: No need for commented out code here, let's remove it

File: app/models/user.rb [hunk 6]
```diff
@@ -1,8 +1,29 @@
...
+         :validatable
+
+  validates :email,
+            uniqueness: true,
+            presence: true
+
+  before_validation :downcase_email
+
+  def profile_complete?
+    if first_name.present? && last_name.present? && weight.present?
```
Comment (lshepstone) [inline app/models/user.rb]: No need for the explicit `if` ... `else` ... `end`, you can simply leave it as one line:

```ruby
def profile_complete?
  first_name.present? && last_name.present? && weight.present?
end
```

Remember, Ruby methods always return the result of the last line.

File: app/forms/user/profile_form.rb [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    :last_name
+  ].freeze
+  # rubocop: enable Style/SymbolArray
+  validates :weight,
+            presence: true
+
+  validates :first_name,
+            :last_name,
+            presence: true,
+            format: { with: /([a-zA-Z])\w+/, message: 'only allows letters' }
```
Comment (lshepstone) [inline app/forms/user/profile_form.rb]: Let's start off on the right foot, and move the string for `message` here to a `config/locales/en.yml` – see https://guides.rubyonrails.org/i18n.html#translations-for-active-record-models

File: app/views/profiles/edit.html.erb [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<div class="row" style="margin-top: 100px">
```
Comment (lshepstone) [inline app/views/profiles/edit.html.erb]: Usually, when you have to use `style` attributes on HTML elements, it indicates that there is a problem with the SCSS approach. For now, remove the `style` attribute here and we will fix up the frontend under the card [#47 Refactor existing SCSS approach](https://trello.com/c/i0LIelil)

File: app/views/devise/registrations/new.html.erb [hunk 9]
```diff
@@ -1,34 +1,44 @@
...
-        <div class="field">
+        <div class="form-group">
           <%= f.label :password_confirmation %><br/>
-          <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+          <%= f.password_field :password_confirmation, autocomplete: "new-password", class: 'form-control' %>
         </div>
 
-        <div class="actions">
-          <%= f.submit "Sign up" %>
+        <div class="form-group" style="margin-top: 20px;">
```
Comment (lshepstone) [inline app/views/devise/registrations/new.html.erb]: See my comment below about `style` attributes

**Comment by lshepstone** [comment]
@tsrandrei I would also encourage you to find any code that has been commented out and remove it – that's a best practice (for example, I saw some `<li>...</li>` markup somewhere that was commented out).

File: app/views/layouts/elements/_dropdown_user_menu.html.erb [hunk 10]
```diff
@@ -11,7 +11,7 @@
 <!--    </ul>-->
 <!--  </li>-->
   <li><a href="#">Dashboard</a></li>
-  <li><a href="#">My Profile</a></li>
+  <li><%= link_to 'My Profile',edit_user_registration_path %></li>
```
[the remark this answers is not in the document]
Comment (tsrandrei) [reply app/views/layouts/elements/_dropdown_user_menu.html.erb]: You've got a hawk eye!

File: config/environments/development.rb [hunk 11]
```diff
@@ -22,17 +22,17 @@
 
   config.action_mailer.perform_deliveries = true
   config.action_mailer.raise_delivery_errors = true
-  config.action_mailer.default_url_options = { host: ENV['MAIL_HOST']}
-  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.default_url_options = { host: ENV['MAIL_HOST'] }
+  config.action_mailer.delivery_method = :letter_opener
   config.action_mailer.perform_caching = false
   config.action_mailer.smtp_settings = {
```
[the remark this answers is not in the document]
Comment (tsrandrei) [reply config/environments/development.rb]: Should ```

**Pull request merged**